core/models.py

```diff
@@ -23,3 +26,23 @@ def get_simple_display_name(self):
     @staticmethod
     def set_display_name_method(method):
         ExtendedUser.display_name_method = method
+
+
+# File path to upload achievement images to
+def get_image_upload_path(instance, filename):
+    # Obtain extension
+    # NB: A file can be renamed to have ANY extension
```

Comment:
Should there be a filter on the allowed file extensions? f.e. only .png, .jpg and .jpeg to prevent people from uploading .js files or other nasty stuff

Reply:
The relevant field is an ImageField which already does so :smile:
Do note that people can rename a file to have any extension though, and I'm not sure how this is handled.